flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/strategies/SpecificTypeStrategies.java

```diff
@@ -58,8 +58,12 @@ public final class SpecificTypeStrategies {
     /** See {@link ArrayTypeStrategy}. */
     public static final TypeStrategy ARRAY = new ArrayTypeStrategy();
 
-    /** See {@link ArrayElementOutputTypeStrategy}. */
-    public static final TypeStrategy ARRAY_ELEMENT = new ArrayElementOutputTypeStrategy();
+    /** Type strategy specific for array element. */
+    public static final TypeStrategy ARRAY_ELEMENT =
+            callContext ->
+                    Optional.of(
+                            ((CollectionDataType) callContext.getArgumentDataTypes().get(0))
+                                    .getElementDataType());
```

Comment:
why you don't use 
```
((CollectionDataType) callContext.getArgumentDataTypes().get(0))
                                    .getElementDataType().nullable);
```

here?

Reply:
@hanyuzheng7 it can be reuse for others.